db/job_test.go

```diff
@@ -1196,44 +968,318 @@ var _ = Describe("Job", func() {
-			BeforeEach(func() {
-				build1, err := job.CreateBuild()
-				Expect(err).NotTo(HaveOccurred())
+	Describe("SyncResourceSpaceCombinations", func() {
+		BeforeEach(func() {
+			otherPipeline, created, err := team.SavePipeline("other-fake-pipeline", atc.Config{
+				Jobs: atc.JobConfigs{
+					{
+						Name: "some-job",
 
-				started, err := build1.Start("some-engine", `{"some":"metadata"}`, atc.Plan{})
-				Expect(err).NotTo(HaveOccurred())
-				Expect(started).To(BeTrue())
-			})
+						Public: true,
 
-			It("creates a build", func() {
-				err := job.EnsurePendingBuildExists()
-				Expect(err).NotTo(HaveOccurred())
+						Serial: true,
 
-				pendingBuilds, err := job.GetPendingBuilds()
-				Expect(err).NotTo(HaveOccurred())
-				Expect(pendingBuilds).To(HaveLen(1))
+						SerialGroups: []string{"serial-group"},
+
+						Plan: atc.PlanSequence{
+							{
+								Put: "some-resource",
+								Params: atc.Params{
+									"some-param": "some-value",
+								},
+							},
+							{
+								Get:      "some-input",
+								Resource: "some-resource",
+								Params: atc.Params{
+									"some-param": "some-value",
+								},
+								Passed:  []string{"job-1", "job-2"},
+								Trigger: true,
+							},
+							{
+								Task:           "some-task",
+								Privileged:     true,
+								TaskConfigPath: "some/config/path.yml",
+								TaskConfig: &atc.TaskConfig{
+									RootfsURI: "some-image",
+								},
+							},
+						},
+					},
+					{
+						Name: "some-other-job",
+					},
+					{
+						Name:         "other-serial-group-job",
+						SerialGroups: []string{"serial-group", "really-different-group"},
+					},
+					{
+						Name:         "different-serial-group-job",
+						SerialGroups: []string{"different-serial-group"},
+					},
+				},
+				Resources: atc.ResourceConfigs{
+					{
+						Name: "some-resource",
+						Type: "some-type",
+					},
+					{
+						Name: "some-other-resource",
+						Type: "some-type",
+					},
+				},
+			}, db.ConfigVersion(0), db.PipelineUnpaused)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(created).To(BeTrue())
+
+			var found bool
+			job, found, err = otherPipeline.Job("some-job")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(found).To(BeTrue())
+
+			tx, err := dbConn.Begin()
+			Expect(err).NotTo(HaveOccurred())
+
+			defer db.Rollback(tx)
+
+			var someResourceID, someOtherResourceID int
+
+			err = psql.Select("id").From("resources").Where(sq.Eq{
+				"resources.pipeline_id": job.PipelineID(),
+				"resources.name":        "some-resource",
+			}).RunWith(tx).QueryRow().Scan(&someResourceID)
+			Expect(err).NotTo(HaveOccurred())
+
+			err = psql.Select("id").From("resources").Where(sq.Eq{
+				"resources.pipeline_id": job.PipelineID(),
+				"resources.name":        "some-other-resource",
+			}).RunWith(tx).QueryRow().Scan(&someOtherResourceID)
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = psql.Insert("resource_spaces").
```

Comment:
I guess we don't have the database methods yet, but it would be preferable to save the resource spaces as if we had discovered them (i.e. using the same methods we'll use in the future for space discovery), rather than writing queries directly.